controllers/configuration/backend/backend.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2022 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backend
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/oam-dev/terraform-controller/api/v1beta2"
+	"github.com/pkg/errors"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/gocty"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var backendInitFuncMap = map[v1beta2.BackendType]*backendInitFunc{
+	"kubernetes": {
+		InitFuncFromHCL:  newK8SBackendFromHCL,
+		InitFuncFromConf: newK8SBackendFromConf,
+	},
+}
+
+type Backend interface {
+	HCL() string
+	GetTFStateJSON(ctx context.Context) ([]byte, error)
+	CleanUp(ctx context.Context) error
+}
+
+type backendInitFunc struct {
+	InitFuncFromHCL  func(string, *ParsedBackendConfig, client.Client) (Backend, error)
+	InitFuncFromConf func(string, interface{}, client.Client) (Backend, error)
+}
+
+type ParsedBackendConfig struct {
+	Name   string   `hcl:"name,label"`
+	Remain hcl.Body `hcl:",remain"`
```

Comment:
Can you write some comments for Name and Remain?

Reply:
Sure. I will.